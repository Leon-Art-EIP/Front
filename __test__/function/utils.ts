import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
