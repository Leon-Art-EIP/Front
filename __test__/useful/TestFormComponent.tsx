import { FormProvider } from "react-hook-form";
import type { ReactNode } from "react";
import useTestForm from "./useTestForm";

interface ITestInputProps {
  name: string;
  value?: string[] | string;
  children: ReactNode;
}

export default function TestFormComponent(props: ITestInputProps): JSX.Element {
  const methods = useTestForm({ [props.name]: props.value });

  return <FormProvider {...methods}>{props.children}</FormProvider>;
}
