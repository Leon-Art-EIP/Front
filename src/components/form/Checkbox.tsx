import { useController } from "react-hook-form";
import { cn } from "../../tools/cn";

interface ICheckboxProps {
  name: string;
  className?: string;
  title?: string;
}

export default function Checkbox(props: ICheckboxProps): JSX.Element {
  const {
    field: { onChange, value },
    fieldState: { error },
  } = useController({ name: props.name });

  return (
    <div className="flex flex-col gap-2 sm:min-w-min min-w-0">
      <div className="flex gap-2 items-center">
        <input
          type="checkbox"
          name={props.name}
          id={props.name}
          className={cn("sm:min-w-min min-w-0 h-4 w-4", props.className)}
          onChange={onChange}
          value={value}
          />
        {props.title && <label className="text-xl">{props.title}</label>}
      </div>
      {error && <div className="text-primary">{error.message}</div>}
    </div>
  );
}
