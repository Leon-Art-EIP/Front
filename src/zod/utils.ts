export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
