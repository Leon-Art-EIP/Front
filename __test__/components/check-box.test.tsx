import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TestFormComponent from "../useful/TestFormComponent";
import Checkbox from "../../src/components/form/Checkbox";

test("Checkbox snapshot", () => {
  testSnapshot(
    <TestFormComponent name="checkbox">
      <Checkbox name="checkbox" />
    </TestFormComponent>
  );
});
