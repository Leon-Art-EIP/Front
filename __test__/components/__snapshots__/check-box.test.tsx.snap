// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox snapshot 1`] = `
<div
  className="flex flex-col gap-2 sm:min-w-min min-w-0"
>
  <div
    className="flex gap-2 items-center"
  >
    <input
      className="sm:min-w-min min-w-0 h-4 w-4"
      id="checkbox"
      name="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
